### PR TITLE
docs: Add examples for GitHub Actions

### PR DESCRIPTION
**Pull request by GuiEpi**
Here is my proposal for the github action in the publishing section of the documentation.

### Jobs
1. `Version`
Checks for changes in the `package.json` file. If there is a change, it retrieves the version of your extension.
3. `Submit`
 Submits your extension to the store using the `wxt submit` command.

`pnpm` and `npm` code-group.

Here's a[ link to my CI Deploy](https://github.com/GuiEpi/plex-skipper/actions/runs/8222114482/job/22483242808) to show you how it works.
I follow the [pnpm continuous integration guide ](https://pnpm.io/continuous-integration#github-actions) and your [publishing](https://wxt.dev/guide/publishing.html) section.

### CODE REVIEW

File: docs/guide/publishing.md [hunk 1]
```diff
@@ -66,10 +66,165 @@ See the [Firefox Addon Store](#firefox-addon-store) section for more details abo
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Build extensions
+        run: |
+          pnpm run build && 
+          pnpm run build:firefox
```
Comment (aklinker1) [inline docs/guide/publishing.md:129]: This step isn't necessary. Extension is built during ZIP.

```suggestion
```

File: docs/guide/publishing.md [hunk 2]
```diff
@@ -66,10 +66,165 @@ See the [Firefox Addon Store](#firefox-addon-store) section for more details abo
...
+          FIREFOX_EXTENSION_ID: ${{ secrets.FIREFOX_EXTENSION_ID }}
+          FIREFOX_JWT_ISSUER: ${{ secrets.FIREFOX_JWT_ISSUER }}
+          FIREFOX_JWT_SECRET: ${{ secrets.FIREFOX_JWT_SECRET }}
+```
+
+```yml [npm]
+env:
+  DIRECTORY: .output
+  PROJECT_NAME: myextension
+
```
Comment (aklinker1) [inline docs/guide/publishing.md:154]: Do you think we need multiple examples for package managers? Or should we just list one?

File: docs/guide/publishing.md [hunk 3]
```diff
@@ -66,10 +66,165 @@ See the [Firefox Addon Store](#firefox-addon-store) section for more details abo
...
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+      - name: Install dependencies
+        run: npm install
+      - name: Build extensions
+        run: |
+          npm run build && 
+          npm run build:firefox
```
Comment (aklinker1) [inline docs/guide/publishing.md:205]:
```suggestion
```

**Review by aklinker1 — CHANGES_REQUESTED** [review]
Looks good. Here's the workflow I was gonna add, for reference: https://github.com/aklinker1/github-better-line-counts/blob/main/.github/workflows/submit.yml

I guess I just forgot to add it? I wrote that a while ago.

It generates a changelog and github release based on conventional commits. I wonder what the best way to document different versioning strategies is?

File: docs/guide/publishing.md (hunk 3 above)
Comment (GuiEpi) [reply docs/guide/publishing.md:201]: Could there be a section called "dispatch" and another called "by commit"?

_Sorry, I made a mistake above when I thought I was commenting on your answer...._

File: docs/guide/publishing.md (hunk 2 above)
Comment (GuiEpi) [reply docs/guide/publishing.md:150]: I did it, but I think it would be better to keep only pnpm.

**Comment by GuiEpi** [comment]
Could there be a section called "dispatch" and another called "by commit"?

**Comment by aklinker1** [comment]
Maybe we just list the two steps that are important (zip and submit), then link to each of our projects for a full workflow?

**Comment by GuiEpi** [comment]
I think it's a good idea, it could look like what I've pushed.

**Review by aklinker1 — APPROVED** [review]
@GuiEpi I updated the wording a little bit, pulled the examples out of the info block, and listed the actual features of the two example workflows so it's more clear what the difference is.

Thanks for adding this for me!

**Pull request merged**